py/test_runner.py

```diff
@@ -482,7 +528,21 @@ def run_test(args):  # pylint: disable=too-many-branches,too-many-statements
         logging.info("Issuing the terminate request")
         for num in range(num_targets):
           full_target = target + "-{0}".format(num)
-          terminateReplica(masterHost, namespace, full_target)
+          terminate_replica(masterHost, namespace, full_target)
+
+      # TODO(richardsliu):
+      # There are lots of verifications in this file, consider refactoring them.
```

Comment:
Yes; I would suggest creating a separate test driver for this test e.g. like
https://github.com/kubeflow/tf-operator/blob/master/py/test_invalid_job.py

But its fine to do this later.